LoopProjectFile/ExtractedInformation.py

```diff
@@ -302,72 +313,151 @@ def SetStratigraphicLog(root, data, append=False, verbose=False):
-        data = []
+        return resp
+    siGroup = resp["value"]
+    data = []
+
+    try:
+        # Get max valid index
         maxValidIndex = min(
             siGroup.dimensions["index"].size, siGroup.getncattr("index_MaxValid")
         )
-        # Select all option
+
+        # Check if variables exist in the group
+        if "thicknessCalculator" not in siGroup.variables or "thicknessCalculatorActiveFlags" not in siGroup.variables:
+            errStr = "Missing 'thicknessCalculator' or 'thicknessCalculatorActiveFlags' in the NetCDF file."
+            if verbose:
+                print(errStr)
+            return {"errorFlag": True, "errorString": errStr}
+
+        # Fetch thickness calculator and active flags
+        thickness_calculator_data = siGroup.variables["thicknessCalculator"][:]
+        thickness_calculator_active_flags = siGroup.variables["thicknessCalculatorActiveFlags"][:]
+
+        # Select all layers
         if (
-            indexList == []
+            not indexList
             and len(indexRange) == 2
-            and indexRange[0] == 0
-            and indexRange[1] == 0
+            and indexRange == (0, 0)
         ):
-            # Select all
-            for i in range(0, maxValidIndex):
-                data.append((siGroup.variables.get("stratigraphicLayers")[i]))
-            response["value"] = data
-        # Select based on list of indices option
-        elif indexList != []:
+            for i in range(maxValidIndex):
+                data.append(siGroup.variables["stratigraphicLayers"][i])
+
+        # Select based on list of indices
+        elif indexList:
             for i in indexList:
-                if int(i) >= 0 and int(i) < maxValidIndex:
-                    data.append((siGroup.variables.get("stratigraphicLayers")[i]))
-            response["value"] = data
-        # Select based on indices range option
+                if 0 <= int(i) < maxValidIndex:
+                    data.append(siGroup.variables["stratigraphicLayers"][i])
+
+        # Select based on index range
         elif (
             len(indexRange) == 2
-            and indexRange[0] >= 0
-            and indexRange[1] >= indexRange[0]
+            and 0 <= indexRange[0] < indexRange[1] <= maxValidIndex
         ):
             for i in range(indexRange[0], indexRange[1]):
-                if int(i) >= 0 and int(i) < maxValidIndex:
-                    data.append((siGroup.variables.get("stratigraphicLayers")[i]))
-            response["value"] = data
+                data.append(siGroup.variables["stratigraphicLayers"][i])
+
         else:
-            errStr = "Non-implemented filter option"
+            errStr = "Invalid filter option for stratigraphic log retrieval."
             if verbose:
                 print(errStr)
-            response = {"errorFlag": True, "errorString": errStr}
+            return {"errorFlag": True, "errorString": errStr}
+
+        # Construct response
+        response["value"] = {
```

Comment:
I've been having a closer look at the code - the new `GetStratigraphicLog` function breaks the `ElementToDataframe` function as the ElementToDataFrame expects a table in the value response not a dictionary of tables. 

I suggest we add a new element to the project file for storing the stratigraphicThicknesses, otherwise we need to change the signature of all of the other getter functions.

Reply:
Thanks for pointing that out. 
Can you please post a minimum reproducible example of this code and error, so I can have a look?

Reply:
for a projectfile from map2loop

```
pf = LPF.OpenProjectFile(loop_project_filename)
LPF.ElementToDataframe(
    loop_project_filename, "stratigraphicLog", LPF.stratigraphicLayerType
)
```